src/EnergyPlus/HVACVariableRefrigerantFlow.cc

```diff
@@ -7920,22 +8413,25 @@ namespace HVACVariableRefrigerantFlow {
                 // Air terminal supply side mixer, calculate supply side mixer output
                 SimATMixer(this->ATMixerName, FirstHVACIteration, this->ATMixerIndex);
                 TempOut = DataLoopNode::Node(ATMixOutNode).Temp;
-                TempIn = DataLoopNode::Node(ZoneNode).Temp;
                 SpecHumOut = DataLoopNode::Node(ATMixOutNode).HumRat;
-                SpecHumIn = DataLoopNode::Node(ZoneNode).HumRat;
-                AirMassFlow = DataLoopNode::Node( ATMixOutNode ).MassFlowRate;
+                AirMassFlow = DataLoopNode::Node(ATMixOutNode).MassFlowRate;
             } else {
                 // Air terminal inlet side mixer
                 TempOut = DataLoopNode::Node(VRFTUOutletNodeNum).Temp;
-                TempIn = DataLoopNode::Node(ZoneNode).Temp;
                 SpecHumOut = DataLoopNode::Node(VRFTUOutletNodeNum).HumRat;
-                SpecHumIn = DataLoopNode::Node(ZoneNode).HumRat;
             }
+            TempIn = DataLoopNode::Node(ZoneNode).Temp;
+            SpecHumIn = DataLoopNode::Node(ZoneNode).HumRat;
         } else {
             TempOut = DataLoopNode::Node(VRFTUOutletNodeNum).Temp;
-            TempIn = DataLoopNode::Node(VRFTUInletNodeNum).Temp;
             SpecHumOut = DataLoopNode::Node(VRFTUOutletNodeNum).HumRat;
-            SpecHumIn = DataLoopNode::Node(VRFTUInletNodeNum).HumRat;
+            if (ZoneNode > 0) {
```

Comment:
This is new, SP controlled TUs woun't have a ZoneNode to be controlled by. Load based control has a ZoneNode and LoadMet (below) is with respect to zone condition. SP control doesn't need a LoadMet indicator and is with respect to the TU inlet node to outlet node.